### PR TITLE
Daos 4240 - Refreshed patch required to include new and revised documentation

### PR DESCRIPTION
**Pull request by cdurf1**
Skip-build: true
Skip-test: true

Updated with changed README files, added initial version of API, reorganized index.md and added new mkdocs.yaml, doxyfile, api html files.
Developer documentation added to the doc nav tree as well as API

### CODE REVIEW

**Comment by johannlombardi** [comment]
So it sounds like those the doc/DEV includes real copy of the files and no symlink.

jlombard@ikoralt-mobl1:~/dev/daos/doc/update/daos$ ls -l doc/Dev/VOS.md
-rw-r--r--  1 jlombard  staff  43737 Mar  4 13:34 doc/Dev/VOS.md
jlombard@ikoralt-mobl1:~/dev/daos/doc/update/daos$ git status doc/Dev/VOS.md
On branch craig/DAOS_4240
Your branch is up-to-date with 'remotes/craig/DAOS_4240'.
nothing to commit, working directory clean

File: doc/overview/ext_tix.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+We need to be responsive to these initial POCs. They need tracked and responded to.
+A process need established to process the request and parse out the tix to responsible EMs and track the tix through the process (Craig).
+Verify closure on tix and appropriate response.
+Triage Channel to prioritize external tix
+Wiki to establish how to submit tix
+HP has been told to email their ticket to Johann directly (blocking only) - General questions are to be mailed to the group (onboarding Wiki for access).
+Can we close the ability to create issues in Github? (Scott K) - Yes. Question is if requesting access to Jira too high a bar? How to tell them how to request an account? Brian can close. Lets wait until we find how difficult requests are to approve. In the interim, monitor github issues.
+Do external users have read access to ALL Jira tix? Is this intended? Private projects will be created at the appropriate time.
+We need codenames for external customers/tix for sortability and to conceal customer identity - NO
+Triage channel to be created. Use the channel to create a rapid response.
```
Comment (johannlombardi) [inline doc/overview/ext_tix.txt]: Not sure what this file is in the documentation ...

Comment (cdurf1) [reply doc/overview/ext_tix.txt]: Nor am I. it should definitely be out.

Comment (cdurf1) [reply doc/overview/ext_tix.txt]: deleted the 3 stray txt files

**Pull request closed without merging**